controllers/movie.js

```diff
@@ -48,25 +54,34 @@ const readByGenre = async (req, res) => {
 
 const readByRecommendation = async (req, res) => {
   console.log(req.body);
+  const { certifications, genres, release, ratings, providers, foreign } = req.body;
 
-  const applicableCerts = recommendations.getCertifications(
-    req.body.certification,
+  // const applicableCerts = recommendations.getCertifications(req.body.certification);
+  const applicableCerts = Object.keys(certifications).filter(
+    certKey => certifications[certKey],
   );
+  const applicableGenres = Object.keys(genres).filter(genreKey => genres[genreKey]);
 
   const conditions = {
-    genres: { $in: req.body.genres },
+    genres: { $in: applicableGenres },
     certification: { $in: applicableCerts },
-    release_year: { $gte: req.body.min_year, $lte: req.body.max_year },
-    'ratings.rotten_tomatoes.value': { $gte: req.body.rotten_tomatoes },
-    'ratings.imdb.value': { $gte: req.body.imdb },
+    release_year: { $gte: release.minYear, $lte: release.maxYear },
+    'ratings.rotten_tomatoes.value': {
+      $gte: ratings.rottenTomatoes.minRating,
+      $lte: ratings.rottenTomatoes.maxRating,
+    },
+    'ratings.imdb.value': { $gte: ratings.imdb.minRating, $lte: ratings.imdb.maxRating },
   };
-
-  if (!req.body.foreign) conditions.original_language = 'en';
-  if (req.body.indie)
-    conditions.budget = { $lte: recommendations.INDIE_BUDGET_THRESHOLD };
+  if (!foreign) conditions.original_language = 'en';
+  // TODO: add provider checking?
+  // if (providers) {
+  //   const applicableProviders = Object.keys(providers).filter(provKey => providers[provKey]);
+  //   conditions['offers.stream[??].provider._id'] = { $in: applicableProviders };
+  // }
 
   const foundMovies = await Movie.paginate(conditions, {
     page: req.query.page || 1,
+    limit: req.body.maxRecs || requests.JW_SEARCH.data.page_size,
   });
   res.json(foundMovies);
```

Comment:
Great code! I would add some error handling though so the server won't crash

Reply:
Yeah, we definitely should add error checking, but there's enough of it necessary that it should probably be its own entire issue.